src/marqo/tensor_search/tensor_search.py

```diff
@@ -1419,14 +1406,17 @@ def bulk_msearch(config: Config, body: List[Dict]) -> List[Dict]:
     except KeyError as e:
         # KeyError indicates we have received a non-successful result
         try:
-            if "index.max_result_window" in response["responses"][0]["error"]["root_cause"][0]["reason"]:
```

Comment:
This just makes it cleaner, and adds 
`elif  root_cause_type == 'query_shard_exception' and root_cause_reason.startswith("Failed to parse query")` to replicate what is in search.

Reply:
good to clean this up 👍